src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -265,4 +267,15 @@ private function getReflectionClass($id, Definition $definition)
             // return null
         }
     }
+
+    private function addServiceToAmbiguousType($id, $type)
+    {
+        // keep an array of all services matching this type
+        if (!isset($this->ambiguousServiceTypes[$type])) {
+            $this->ambiguousServiceTypes[$type] = array(
+                $this->types[$type],
```

Comment:
Yeah, but `addServiceToAmbiguousType()` is also called before that check: https://github.com/weaverryan/symfony/blob/autowire-colission-fix/src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php#L186